internal/cortex/querier/queryrange/instrumentation.go

```diff
@@ -12,19 +12,51 @@ import (
 	"github.com/weaveworks/common/instrument"
 )
 
+const DAY = 24 * time.Hour
+const QUERYDURATIONBUCKET = "query_range_bucket"
+
 // InstrumentMiddleware can be inserted into the middleware chain to expose timing information.
 func InstrumentMiddleware(name string, metrics *InstrumentMiddlewareMetrics) Middleware {
-	var durationCol instrument.Collector
-
-	// Support the case metrics shouldn't be tracked (ie. unit tests).
-	if metrics != nil {
-		durationCol = instrument.NewHistogramCollector(metrics.duration)
-	} else {
-		durationCol = &NoopCollector{}
-	}
 
 	return MiddlewareFunc(func(next Handler) Handler {
 		return HandlerFunc(func(ctx context.Context, req Request) (Response, error) {
+
+			var durationCol instrument.Collector
```

Comment:
why you move this into the HandlerFunc?

Reply:
because it requires the request parameter to extract the start and end information